### ISSUE TITLE
Great Demo, Thank you

### ISSUE THREAD

**Issue by waelsaad** [Alexsays/ASAppleMusic#2]
I've learn't a lot from this, thank you it came to me at the right time, there is still room for improvements.  If there was a better way to make callWithToken a bit more generic and avoid the repeated lines of code for each entity that would be awesome but seriously thank you.

**Comment by Alexsilvacodes** [comment]
Yep, it was done a bit on a rush, there's a lot of work in refactoring code, etc, but I'm glad that you find it useful! If you want to make some changes I'm totally open for it :D

**Comment by waelsaad** [comment]
One more important thought as I am now integrating it in a project I am working on. I believe all these classes should be Decodable and remove Alamofire and EVReflection, the implementation will be cleaner and simpler, I understand there is an effort to do this but it will be better hopefully you will actually consider when you have time. Thanks

**Comment by Alexsilvacodes** [comment]
I was thinking about it the other day. Now I have a bit more time, I will start working on it. Thanks for the suggestion!

**Comment by waelsaad** [comment]
Excellent and thank you for the prompt response and considering it, I'll be watching the branch :)

**Comment by Alexsilvacodes** [comment]
I started working right now adapting the API to Codable/Decodable, in fact it should be working more or less fine, I'm just adapting the Sample code and trying to create Unit tests. You can take a look in `codable` branch
